### PR TITLE
Add layout for settings redesign

### PR DESCRIPTION
**Pull request by flozia**
# References:

Jira: [MNTOR-3675](https://mozilla-hub.atlassian.net/browse/MNTOR-3675)
Figma: https://www.figma.com/design/2iCgADpFXKOFZTEgCPkQsR/Settings?node-id=6-17580&t=uQL3qalyCpHOWgWx-4

# Description

Adds the new layout for the settings page redesign and is behind the feature flag `SettingsPageRedesign`. The panel content will be implemented separately:

- Edit your info: [MNTOR-3803](https://mozilla-hub.atlassian.net/browse/MNTOR-3803)
- Set notifications: [MNTOR-3676](https://mozilla-hub.atlassian.net/browse/MNTOR-3676)
- Manage account: [MNTOR-3677](https://mozilla-hub.atlassian.net/browse/MNTOR-3677)

# Screenshot

### Panels

| Edit your info | Set notifications | Manage account |
| --- | --- | --- |
| image | image | image |

### Mobile menu

image

# How to test

1. Enable the feature flag `SettingsPageRedesign`.
2. Visit the settings page `/user/settings`.

# Checklist (Definition of Done)

- [x] Localization strings (if needed) have been added.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added or updated the relevant sections in readme and/or code comments
- [x] I've added a unit test to test for potential regressions of this bug.
- [x] If this PR implements a feature flag or experimentation, the Ship Behind Feature Flag status in Jira has been set
- [x] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.

[MNTOR-3675]: https://mozilla-hub.atlassian.net/browse/MNTOR-3675?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MNTOR-3803]: https://mozilla-hub.atlassian.net/browse/MNTOR-3803?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MNTOR-3676]: https://mozilla-hub.atlassian.net/browse/MNTOR-3676?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MNTOR-3677]: https://mozilla-hub.atlassian.net/browse/MNTOR-3677?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by flodolo — APPROVED** [review]

File: src/app/tokens.scss [hunk 1]
```diff
@@ -170,6 +170,10 @@ $border-radius-xl: 32px;
 $border-focus-width: 3px;
 
 // Drop shadow
+$box-shadow-xs:
+  0 4px 6px 1px rgba(29, 17, 51, 0.04),
+  0 1px 8px 1px rgba(9, 32, 77, 0.12),
+  0 3px 5px -1px rgba(29, 17, 51, 0.12);
```
[the remark this answers is not in the document]
Comment (flozia) [reply src/app/tokens.scss:176]: Yes, I wanted to comment on your PR as well. I’ll take over your `$box-shadow-xs` value. :)

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/panels/SettingsPanelEditInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+function SettingsPanelEditInfo() {
+  return <div>Edit info content</div>;
+}
+
+export { SettingsPanelEditInfo };
```
[the remark this answers is not in the document]
Comment (flozia) [reply src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/panels/SettingsPanelEditInfo.tsx:9]: Yes! This is just a placeholder. The content is being introduced in https://github.com/mozilla/blurts-server/pull/5342.

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/View.tsx [hunk 3]
```diff
@@ -59,180 +65,186 @@ export const SettingsView = (props: Props) => {
...
+                  <h3>{l10n.getString("settings-cancel-plus-title")}</h3>
+                  <p>{l10n.getString("settings-cancel-plus-details")}</p>
+                  {props.enabledFeatureFlags.includes("CancellationFlow") ? (
+                    <CancelFlow
+                      enableDiscountCoupon={props.enabledFeatureFlags.includes(
+                        "DiscountCouponNextThreeMonths",
+                      )}
+                      fxaSubscriptionsUrl={props.fxaSubscriptionsUrl}
+                      experimentData={props.experimentData}
+                      isMonthlySubscriber={props.isMonthlySubscriber}
```
[the remark this answers is not in the document]
Comment (flozia) [reply src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/View.tsx:129]: I’ve opened a separate [PR ](https://github.com/mozilla/blurts-server/pull/5342) for introducing the content for the settings panels.

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/SettingsContent.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+  useEffect(() => {
+    const nextPathname = `/user/settings/${activeTab}`;
+    if (pathname !== nextPathname) {
+      // Directly interacting with the history API is recommended by Next.js to
+      // avoid re-rendering on the server:
+      // See https://github.com/vercel/next.js/discussions/48110#discussioncomment-7563979.
+      window.history.replaceState(null, "", nextPathname);
+    }
+  }, [pathname, activeTab]);
```
[the remark this answers is not in the document]
Comment (flozia) [reply src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/SettingsContent.tsx:73]: Added a comment in [e79aa0e](https://github.com/mozilla/blurts-server/pull/5327/commits/e79aa0e9061cf4e1fe79e2c5fe3b99f624054fec).

File: src/app/(proper_react)/(redesign)/MobileShell.module.scss [hunk 5]
```diff
@@ -134,6 +134,32 @@
             outline: none;
           }
         }
+
+        .subMenu {
```
[the remark this answers is not in the document]
Comment (flozia) [reply src/app/(proper_react)/(redesign)/MobileShell.module.scss:138]: Updated the breakpoints in https://github.com/mozilla/blurts-server/pull/5342.

**Review by rhelmer — APPROVED** [review]

**Comment by flozia** [comment]
@rhelmer The changes introduced in commit [6da287e](https://github.com/mozilla/blurts-server/pull/5327/commits/6da287eaf7b1b30337088b76733f041454b1fe73) are from the approved PR #5342. Merging the PR needed some resolution.

**Review by rhelmer — APPROVED** [review]

**Pull request merged**